firefox-ios/Client/Frontend/NativeErrorPage/NativeErrorPageViewController.swift

```diff
@@ -328,13 +328,15 @@ final class NativeErrorPageViewController: UIViewController,
 
     @objc
     private nonisolated func didTapReload() {
```

Comment:
This is marked `nonisolated` but it's called from a `UIButton` tap. In other places we did not mark those UI button  `@objc` selector methods as `nonisolated`, and we call directly `store.dispatch`. It's assumed since the click is from a button and it's in the UI, then it's always the main thread. 

We can totally change that, but I just want us to have consistency.

Reply:
I think this is the developer's responsibility to check for that, rather than for us to plan for every use cases ahead. I am fine with a bit of risk, and prefer the simplicity of not having to call `ensureMainThread` everywhere (as we have a lot of those @objc methods for `UIButton`s that calls redux actions). I would say we can bring that in the iOS engineering meeting, bring awareness to other developers, and if it becomes a problem that often people call `@objc` method from notifications later on, then we can find a way to enforce that all @objc method needs to be `nonisolated`. But I would like for us to try the simple path beforehand, as we might be trying to solve a problem that won't be a problem necessarily? We can always come back and readjust when we have more information!